app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,308 @@
+            if (result.isSuccess) {
+                val agendaDayDTO = result.getOrNull()
+                if (agendaDayDTO != null) {
+                    val events = agendaDayDTO.events.map { it.toDomain() }
+                    val tasks = agendaDayDTO.tasks.map { it.toDomain() }
+                    val reminders = agendaDayDTO.reminders.map { it.toDomain() }
+                    Result.success(events + tasks + reminders)
+                } else {
+                    Result.failure(Exception("Response body is null"))
+                }
+            } else {
+                Result.failure(Exception("Error getting agenda"))
+            }
+        } catch (e: Exception) {
+            Result.failure(Exception("Error getting agenda"))
+        }
+    }
+
+    override suspend fun updateLocalAgendaForDayFromRemote(dateTime: ZonedDateTime) {
+        CoroutineScope(Dispatchers.IO).launch {
+            try {
+                // Get fresh data
+                val result = agendaApi.getAgenda(dateTime)
+
+                if(result.isSuccess) {
+                    val agenda = result.getOrNull()
+
+                    eventRepository.clearEventsForDayLocally(dateTime)
+                    // Insert fresh data locally
+                    agenda?.events?.forEach { event ->
+                        val result2 =
+                            eventRepository.upsertEventLocally(event.toDomain())
+                        if(result2 is ResultUiText.Error) {
+                            throw IllegalStateException(result2.message.asStrOrNull())
+                        }
+                    }
+
+                    taskRepository.clearTasksForDayLocally(dateTime)
+                    // Insert fresh data locally
+                    agenda?.tasks?.forEach { task ->
+                        val result2 =
+                            taskRepository.upsertTaskLocally(task.toDomain())
+                        if(result2 is ResultUiText.Error) {
+                            throw IllegalStateException(result2.message.asStrOrNull())
+                        }
+                    }
+
+                    reminderRepository.clearRemindersForDayLocally(dateTime)
+                    // Insert fresh data locally
+                    agenda?.reminders?.forEach { reminder ->
+                        val result2 =
+                            reminderRepository.upsertReminderLocally(reminder.toDomain())
+                        if(result2 is ResultUiText.Error) {
+                            throw IllegalStateException(result2.message.asStrOrNull())
+                        }
+                    }
+
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+                // don't send error to user, just log it (silent fail is ok here)
+            }
+        }
+    }
+
+    override fun getAgendaForDayFlow(dateTime: ZonedDateTime): Flow<List<AgendaItem>> {
+
+        return flow {
+            supervisorScope {
```

Comment:
This is not needed here, since you're not launching child coroutines

Reply:
ok i will refactor